taylorrgriffin/secret-saver
# Post commit functionality

This branch contains functional pre and post commit hooks, that provide the core behavior. The changes include more robust logging in the hooks, methods to stage the post-commit hook, and the post-commit functionaliy itself via the hook.

Steps to test:

1.  Delete any existing current pre or post hooks in the .git directory
     `rm .git/hooks/pre-commit`
     `rm .git/hooks/post-commit`
2. Delete any existing secrets.json
     `rm secrets.json`
3. Run protect function
     `node index.js`
4. Create a new file that includes the secrets
     `echo "var sec1 = 123456; var sec2 = 'ABCDEF';" > newfile.js`
5. Stage file
     `git add newfile.js`
6. Commit new file
     `git commit -m "Testing core functionality"`
7. Observe logged information from pre and post commit hooks, make sure things seem reasonable
8. Push commit to GitHub so the committed file can be viewed
     `git push origin post-commit-functionality`
9. Ensure that the version of newfile.js on GitHub has the secret hash instead of the secret value
10. Ensure the local copy of newfile.js has the secret value instead of the hash
11. Keep the repo clean by getting rid of newfile
     `rm newfile.js`
     `git add newfile.js`
     `git commit -m "Removing test file"`
     `git push origin post-commit-functionality`

Note: currently only ONE of the two secrets in newfile.js will be changed to the hash on the remote copy, the other one will still be the secret value. There seems to be a bug in the method pushSecretToExistingObj, because currently when you run `node index.js`, only one of the two protected secrets will be added to secrets.json, and it's whichever one is added last. Given that this has nothing to do with the post-commit functionality, and the fact that I've already made a lot of changes in this branch, I'd like to address that bug in a new branch as I plan to completely refactor the pushSecretToExistingObj method.
